packages/theme-check-common/src/checks/valid-local-blocks/index.spec.ts

```diff
@@ -0,0 +1,226 @@
+      {% schema %}
+      {
+        "name": "Section name",
+        "blocks": [
+          {
+            "type": "local_block",
+            "name": "Local Block"
+          }
+        ],
+        "presets": [
+          {
+            "name": "Default",
+            "blocks": [
+              {
+                "type": "text"
+              }
+            ]
+          }
+        ]
+      }
+      {% endschema %}
+    `,
+  };
+
+  const offenses = await check(theme, [ValidLocalBlocks]);
+  console.log(offenses);
+  expect(offenses).to.have.length(1);
+  expect(offenses[0].message).to.equal(
+    'Sections cannot use theme blocks together with locally scoped blocks.',
+  );
+});
+
+it('should report errors when sections use static theme blocks together with locally scoped blocks', async () => {
+  const theme: MockTheme = {
+    'sections/local-blocks.liquid': `
+          {% schema %}
+          {
+            "name": "Section name",
+            "blocks": [
+              {
+                "type": "local_block",
+                "name": "Local Block"
+              }
+            ],
+            "presets": [
+              {
+                "name": "Default",
+                "blocks": [
+                  {
+                    "id": "static_block",
+                    "type": "static_block",
+                    "static": true
+                  }
+                ]
+              }
+            ]
+          }
+          {% endschema %}
+        `,
+  };
+  const offenses = await check(theme, [ValidLocalBlocks]);
+  expect(offenses).to.have.length(1);
+  expect(offenses[0].message).to.equal(
+    'Sections cannot use static theme blocks together with locally scoped blocks.',
+  );
+});
+
+it('should report errors with correct indicies when sections use static theme blocks together with locally scoped blocks', async () => {
+  const theme: MockTheme = {
+    'sections/local-blocks.liquid': `
+            {% schema %}
+            {
+              "name": "Section name",
+              "blocks": [
+                {
+                  "type": "local_block",
+                  "name": "Local Block"
+                }
+              ],
+              "presets": [
+                {
+                  "name": "Default",
+                  "blocks": [
+                    {
+                      "id": "static_block",
+                      "type": "static_block",
+                      "static": true
+                    }
+                  ]
+                }
+              ]
+            }
+            {% endschema %}
+          `,
+  };
+  const offenses = await check(theme, [ValidLocalBlocks]);
+  expect(offenses).to.have.length(1);
+
+  const content = theme['sections/local-blocks.liquid'];
+  const errorContent = content.slice(offenses[0].start.index, offenses[0].end.index);
+  expect(errorContent).to.equal('"static_block"');
+});
+
+it('should report errors when static theme blocks have a name property', async () => {
```

Comment:
If this error is coming from the JSON schema, I wouldn't add this test just because if ever we change something in the `theme-liquid-docs` repo, this test could break.

Reply:
These tests are all testing for `ValidLocalBlocks` specific checks.